### PR TITLE
Add support to verify a signed message

### PR DESCRIPTION
**Pull request by 0xmaayan**
Add function `signMessageAndVerify` in `WalletCore` to verify the signature  and prove a user controls a given private key.

### CODE REVIEW

File: packages/wallet-adapter-core/src/WalletCore.ts [hunk 1]
```diff
@@ -307,4 +310,36 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
       throw new WalletNetworkChangeError(errMsg).message;
     }
   }
+
+  async signMessageAndVerify(message: SignMessagePayload): Promise<boolean> {
+    try {
+      this.isWalletExists();
```
Comment (hariria) [inline packages/wallet-adapter-core/src/WalletCore.ts]: Might be good to rename to doesWalletExist or walletExists

File: packages/wallet-adapter-core/src/WalletCore.ts [hunk 2]
```diff
@@ -307,4 +310,36 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
...
+      if (!this._account) throw new Error("No account found!");
+      const response = await this._wallet?.signMessage(message);
+      if (!response)
+        throw new WalletSignMessageAndVerifyError("Failed to sign a message")
+          .message;
+      // Verify that the bytes were signed using the private key that matches the known public key
+      let verified = false;
+      // Remove the 0x prefix
+      const currentAccountPublicKey = this._account.publicKey.slice(2, 66);
```
Comment (hariria) [inline packages/wallet-adapter-core/src/WalletCore.ts]: What if they don't have hex prefix?

Comment (hariria) [reply packages/wallet-adapter-core/src/WalletCore.ts]: I think there's a verify method on AccountAddress in Aptos package

**Review by hariria — APPROVED** [review]
LGTM some nits

File: packages/wallet-adapter-core/src/WalletCore.ts (hunk 2 above)
Comment (0xmaayan) [reply packages/wallet-adapter-core/src/WalletCore.ts]: thanks! I think there is a WIP validate method to add to the ts sdk, but was able to find a way to ensure this is a hex string and then remove its prefix to use it in the signature.verify method

**Pull request merged**